From: David Hildenbrand <david@redhat.com>
Subject: mm-huge_memory-improve-split_huge_page_to_list_to_order-return-value-documentation-v2
Date: Mon, 22 Apr 2024 21:42:17 +0200

* Also document concurrent removal from the page cache (likely we should
  return -EBUSY -- but likely it doesn't really matter).

* Reference fd4a7ac32918 in patch description

Link: https://lkml.kernel.org/r/20240422194217.442933-1-david@redhat.com
Signed-off-by: David Hildenbrand <david@redhat.com>
Reviewed-by: Zi Yan <ziy@nvidia.com>
Reviewed-by: John Hubbard <jhubbard@nvidia.com>
Cc: Andrew Morton <akpm@linux-foundation.org>
Cc: John Hubbard <jhubbard@nvidia.com>
Cc: Matthew Wilcox <willy@infradead.org>
Cc: Baolin Wang <baolin.wang@linux.alibaba.com>

Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/huge_memory.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/mm/huge_memory.c~mm-huge_memory-improve-split_huge_page_to_list_to_order-return-value-documentation-v2
+++ a/mm/huge_memory.c
@@ -2975,7 +2975,8 @@ bool can_split_folio(struct folio *folio
  *
  * Returns 0 if the huge page was split successfully.
  *
- * Returns -EAGAIN if the folio has unexpected reference (e.g., GUP).
+ * Returns -EAGAIN if the folio has unexpected reference (e.g., GUP) or if
+ * the folio was concurrently removed from the page cache.
  *
  * Returns -EBUSY when trying to split the huge zeropage, if the folio is
  * under writeback, if fs-specific folio metadata cannot currently be
_
